### PR TITLE
Address ref NULL issue which fails API refresh-chromebot-status

### PR DESCRIPTION
**Pull request by keyonghan**
This PR fixes: https://github.com/flutter/flutter/issues/61873

### CODE REVIEW

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart [hunk 1]
```diff
@@ -74,6 +74,31 @@ void main() {
       expect(task.status, Task.statusNew);
     });
 
+    test('do not update task status when commitSha/ref is unknown', () async {
+      final Commit commit = Commit(key: config.db.emptyKey.append(Commit, id: 'abc'), sha: 'abc');
```
Comment (godofredoc) [inline app_dart/test/request_handlers/refresh_chromebot_status_test.dart:78]: Here and below please add a "," at the end of the function calls to format the blocks in a more readable way.

File: app_dart/lib/src/service/luci.dart [hunk 2]
```diff
@@ -68,7 +68,8 @@ class LuciService {
       final LuciBuilder builder = builders.singleWhere((LuciBuilder builder) {
         return builder.name == build.builderId.builder;
       });
-      final BranchLuciBuilder branchLuciBuilder = BranchLuciBuilder(luciBuilder: builder, branch: ref.split('/')[2]);
+      final String branch = ref == 'unknown' ? 'unknown' : ref.split('/')[2];
+      final BranchLuciBuilder branchLuciBuilder = BranchLuciBuilder(luciBuilder: builder, branch: branch);
```
Comment (godofredoc) [inline app_dart/lib/src/service/luci.dart:72]: ',' at the end for better formatting.

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/test/request_handlers/refresh_chromebot_status_test.dart]: Done.

File: app_dart/lib/src/service/luci.dart (hunk 2 above)
Comment (keyonghan) [reply app_dart/lib/src/service/luci.dart]: Done.

**Pull request merged**